clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -592,11 +592,14 @@ private void handleListOffsetResponse(Map<TopicPartition, Long> timestampsToSear
                 log.debug("Cannot search by timestamp for partition {} because the message format version " +
                         "is before 0.10.0", topicPartition);
                 timestampOffsetMap.put(topicPartition, null);
-            } else if (error == Errors.NOT_LEADER_FOR_PARTITION
-                    || error == Errors.UNKNOWN_TOPIC_OR_PARTITION) {
+            } else if (error == Errors.NOT_LEADER_FOR_PARTITION) {
                 log.debug("Attempt to fetch offsets for partition {} failed due to obsolete leadership information, retrying.",
                         topicPartition);
                 future.raise(error);
+            } else if (error == Errors.UNKNOWN_TOPIC_OR_PARTITION) {
+                log.warn("Received unknown topic or partition error in ListOffset request for partition {}. The topic/partition " +
+                        "may not exist or the user may not have Describe access to it", topicPartition);
```

Comment:
Yeah, I was torn on some of these. I'm fine bumping them down to DEBUG level, but I thought we should make some effort to make the problem more visible since this could now be a persistent problem (if the user does not have Describe access).